src/pipeline.ts

```diff
@@ -489,14 +551,483 @@ export async function scanRssFeeds() {
+				);
+				// result is only SUCCESS or FAILURE here but still log original injectionResult
+				if (result === InjectionResult.SUCCESS) {
+					logger.info({
+						label: Label.INJECT,
+						message: `${progress} Injected ${filePathLog} using stalled source, you will need to resume or remove from client - ${chalk.green(injectionResult)}`,
+					});
+					inClient = true;
+					injected = true;
+				} else {
+					logger.warn({
+						label: Label.INJECT,
+						message: `${progress} Failed to inject ${filePathLog} using stalled source - ${chalk.yellow(injectionResult)}`,
+					});
+				}
+			} else {
+				logger.warn({
+					label: Label.INJECT,
+					message: `${progress} Failed to link files for ${filePathLog}, ${linkedFilesRootResult.unwrapErr()} - ${chalk.yellow(injectionResult)}`,
+				});
+			}
+		}
+	}
+	if (inClient && !injected) {
+		if (linkedNewFiles) {
+			logger.info({
+				label: Label.INJECT,
+				message: `${progress} Rechecking ${filePathLog} as new files were linked - ${chalk.green(injectionResult)}`,
+			});
+			await getClient().recheckTorrent(meta.infoHash);
+		} else {
+			logger.warn({
+				label: Label.INJECT,
+				message: `${progress} No new files linked for ${filePathLog}, resume or remove from client - ${chalk.yellow(injectionResult)}`,
+			});
+		}
+	}
+	return injected;
+}
+
+function injectSummary(summary: InjectSummary): void {
+	const incompleteMsg = `${chalk.bold.yellow(summary.ALREADY_EXISTS)} existed in client${
+		summary.INCOMPLETE_CANDIDATES
+			? chalk.dim(` (${summary.INCOMPLETE_CANDIDATES} were incomplete)`)
+			: ""
+	}`;
+	const resultMsg = formatAsList(
+		[
+			`Injected ${chalk.bold.green(summary.INJECTED)}/${chalk.bold.white(summary.TOTAL)} torrents`,
+			summary.FULL_MATCHES &&
+				`${chalk.bold.green(summary.FULL_MATCHES)} were full matches`,
+			summary.PARTIAL_MATCHES &&
+				`${chalk.bold.yellow(summary.PARTIAL_MATCHES)} were partial matches`,
+			summary.INCOMPLETE_SEARCHEES &&
+				`${chalk.bold.yellow(summary.INCOMPLETE_SEARCHEES)} had incomplete sources`,
+			summary.ALREADY_EXISTS && incompleteMsg,
+			summary.BLOCKED &&
+				`${chalk.bold.yellow(summary.BLOCKED)} were possibly blocklisted`,
+			summary.FAILED &&
+				`${chalk.bold.red(summary.FAILED)} failed to inject`,
+			summary.UNMATCHED &&
+				`${chalk.bold.red(summary.UNMATCHED)} had no matches`,
+		].filter(isTruthy),
+		{ sort: false, type: "unit" },
+	);
+	logger.info({ label: Label.INJECT, message: chalk.cyan(resultMsg) });
+
+	if (summary.UNMATCHED > 0) {
+		logger.info({
+			label: Label.INJECT,
+			message: `Use "${chalk.bold.white("cross-seed diff")}" to get the reasons two torrents are not considered matches`,
+		});
+	}
+
+	if (summary.FOUND_BAD_FORMAT && !summary.FLAT_LINKING) {
+		logger.warn({
+			label: Label.INJECT,
+			message: `Some torrents could be linked to linkDir/${UNKNOWN_TRACKER} - follow .torrent naming format in the docs to avoid this`,
+		});
+	}
+}
+
+export async function injectSavedTorrents() {
```

Comment:
Good work breaking down this function so far. I'm still finding it somewhat hard to read because it's 215 lines long, there are many variables in scope, and there's a fair amount of mutation. Let's try to get it under 100. 

I would start by extracting the body of the `for (const torrentFilePath of dirContents)` loop into its own function. (this has precedent, with `performAction`/`performActions` and `findOnOtherSites`/`findMatchesBatch`). 

From there, I think extracting the bodies of the `TORRENT_NOT_COMPLETE` and `ALREADY_EXISTS` branches in the aftermath of `injectInitialAction` would make sense. We can pass around the `summary` object if needed.

Let's put all of these functions (really, all of the functions added to `pipeline.ts` in this PR) into a new file, `injectSavedTorrents.ts` (or `inject.ts`, if you prefer.) These are pretty independent codepaths from the other code in `pipeline.ts`.

Removing the `toDelete` builder code should help as well.